nifi-extension-bundles/nifi-aws-bundle/nifi-aws-processors/src/main/java/org/apache/nifi/processors/aws/credentials/provider/service/AWSCredentialsProviderControllerService.java

```diff
@@ -260,6 +263,12 @@ public class AWSCredentialsProviderControllerService extends AbstractControllerS
         .dynamicallyModifiesClasspath(true)
         .build();
 
+    public static final PropertyDescriptor OAUTH2_ACCESS_TOKEN_PROVIDER = new PropertyDescriptor.Builder()
+        .name("OAuth2 Access Token Provider")
+        .description("Controller Service providing OAuth2/OIDC tokens to exchange for AWS temporary credentials using STS AssumeRoleWithWebIdentity.")
+        .identifiesControllerService(OAuth2AccessTokenProvider.class)
+        .required(false)
```

Comment:
What options are available to make this dependent on other properties? Perhaps the `Assume Role ARN`? It may also be helpful to introduce some type of strategy property where `WebIdentity` is one of several allowed values, on which this property would depend. That should help make it clear when this service is actually needed.

Reply:
Yeah adding `.dependsOn(ASSUME_ROLE_ARN)`. I did think about adding a strategy property, but it was not obvious how to make it work and be 100% sure that it would not break existing flows during an upgrade. We have quite a lot of possible combinations here.